33833 implement rest api get user permissions

hassandotcms opened the pull request:
### Proposed Changes
- Add GET /api/v1/permissions/user/{userId} endpoint to PermissionResource for retrieving a user's permission assignments organized by assets (hosts/folders) with pagination support
- Add comprehensive test coverage including integration tests and Postman tests for response structure, email lookup, pagination, and error handling

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

This PR fixes: #33833

This PR fixes: #33833

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- add new response view class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
- implement helper methods for user permission rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
 - get api implementation to return user specific permissions by id/email, by hosts/folder.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- integration test added to verify the functionality of get user permissions rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

add postman test.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

fix integration tests, response type error.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

support api classes to be available for injection via CDI.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

enable CDI for userPermissionHelper, to enable injection.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 fix integ tests

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

  - Return specific response type instead of generic Response
  - Add better documentation and error messages
  - Split user loading into separate helper method

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 refactoring and cleanup.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

move helper method to helper class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 add view class for permission metadata.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

implement rest api for permission metadata.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

add javadoc, refactoring in helper methods to make them reusable and update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 fix constructor init issue.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 add postman test for permission metadata api

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(users-portlet) #33394

 - Create PermissionMetadata, UserPermissions, and UserPermissionAsset beans for type-safe API responses
 - Update permission endpoints to return ResponseEntityView wrappers instead of generic Response objects
 - Replace Map-based data structures with beans

hassandotcms committed:
modernize(users-portlet): rest api for saving permissions [#33393]

- created rest resource endpoint to save/update user permissions.
- helper methods to validate transform permission data

hassandotcms committed:
modernize(users-portlet): rest api for saving permissions [#33393]

- write integration tests to validate the user permissions.
  success/validation scenarios.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
Merge branch '33394-implement-rest-api-get-permission-metadata' into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(users-portlet): rest api for permission metadata #33394

update auth for permission api.

hassandotcms committed:
Merge branch 'main' into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(users-portlet): rest api to save user permissions #33393

 1. Change List→Set for permission levels to prevent duplicates (PermissionMetadataView, SaveUserPermissionsForm, UserPermissionAsset, UserPermissionHelper)
  2. Rename PermissionMetadata→PermissionMetadataView following dotCMS naming conventions for REST views
  3. Replace switch statement with Function map in convertPermissionNamesToBits(), rename systemUser→user parameter

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
refactor(permissions): align metadata API with specification #33394

- Change endpoint path from /metadata to / (root collection path)
  - Update permission scope terminology from STRUCTURE to CONTENT_TYPE
  - Add 403 Forbidden response documentation to OpenAPI annotations
  - Update Postman collection to use new endpoint path

hassandotcms committed:
api doc update

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch '33394-implement-rest-api-get-permission-metadata' into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): refactor user permissions API #33393

  Added PermissionResourceIntegrationTest with tests for GET/PUT endpoints.
  Updated PermissionResourceIntegrationTest and Postman collection to use new paths.

hassandotcms committed:
modernize(permissions): add paginated GET endpoint for user permissions along with user object #33833

hassandotcms committed:
modernize(permissions): 33833- remove extra file

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): rest api to update user permisisons #33393

sync with main and refactoring of file organization and tests.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
sync main

hassandotcms committed:
modernize(permissions): rest api to update user permissions (#33393)

fix failing tests.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch '33393-implement-rest-api-save-user-permissions-endpoint' into 33833-implement-rest-api-get-user-permissions

hassandotcms committed:
modernize(permission): rest api to get user permissions [#33833]

1. fix failing postman tests expecting userId instead of user.
2. fix constructor init issue in integration tests.
3. refactor code to move files from user package to permission package.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33833-implement-rest-api-get-user-permissions

hassandotcms committed:
modernize(permissions): rest api to get permissions of user. #33833

- remove unused files.

hassandotcms committed:
modernize(permissions): rest api to get user permissions #33833

- use immutables.

hassandotcms committed:
modernize(permissions): rest api to get user permissions #33833

- Fix integration test method names to use Immutables-style accessors (cascadeInitiated(), asset(), userId()) instead of JavaBean-style getters
- Add Postman tests for GET user permissions endpoint covering response structure, email lookup, pagination, 404 handling, and asset validation

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33833-implement-rest-api-get-user-permissions

hassandotcms committed:
add user id to log statement #33833

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33833-implement-rest-api-get-user-permissions

fabrizzio-dotCMS commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/system/permission/AbstractSaveUserPermissionsView.java:
```diff
@@ -0,0 +1,71 @@
+@JsonSerialize(as = SaveUserPermissionsView.class)
+@JsonDeserialize(as = SaveUserPermissionsView.class)
+@Schema(description = "Result of saving user permissions on an asset")
+public interface AbstractSaveUserPermissionsView {
```
It's a convention we use with Immutables 
see @Value.Style(typeImmutable = "*", typeAbstract = "Abstract*")

hassandotcms committed:
feedback: get user permissions api #33833

 1. PermissionResource.java - Replaced manual pagination logic with PaginationUtil.getPageView() using withFunction() to build complex UserPermissionsView response. Now returns
  ResponseEntityPaginatedDataView and gets all 5 pagination headers automatically.
  2. PermissionResourceIntegrationTest.java - Updated GET tests to cast response.getEntity() to UserPermissionsView

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33833-implement-rest-api-get-user-permissions

fmontes committed:
Merge branch 'main' into 33833-implement-rest-api-get-user-permissions
